packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts

```diff
@@ -63,16 +36,76 @@ const selectors = {
 export class EditorToolbarComponent {
 	private page: Page;
 	private editor: Locator;
+	private selectors: Selectors;
 
 	/**
 	 * Constructs an instance of the component.
 	 *
 	 * @param {Page} page The underlying page.
 	 * @param {Locator} editor Locator or FrameLocator to the editor.
+	 * @param {string} context Context of the editor (e.g. whether it's editing a post or the site).
 	 */
-	constructor( page: Page, editor: Locator ) {
+	constructor(
+		page: Page,
+		editor: Locator,
+		context: 'post-editor' | 'site-editor' = 'post-editor'
+	) {
 		this.page = page;
 		this.editor = editor;
+
+		const panel = context === 'post-editor' ? postEditorPanel : siteEditorPanel;
+		this.selectors = {
+			panel: panel,
+			// Block Inserter
+			// Note the partial class match. This is to support site and post editor. We can't use aria-label because of i18n. :(
+			blockInserterButton: `${ panel } button[class*="inserter-toggle"]`,
+
+			// Draft
+			saveDraftButton: `${ panel } button[aria-label="Save draft"], button[aria-label="Saved"]`,
+			switchToDraftButton: `${ panel } button.editor-post-switch-to-draft`,
+
+			// Preview
+			previewButton: `${ panel } :text("View"):visible`,
+			desktopPreviewMenuItem: ( target: EditorPreviewOptions ) =>
+				`button[role="menuitem"] span:text("${ target }")`,
+			previewPane: ( target: EditorPreviewOptions ) => `.is-${ target.toLowerCase() }-preview`,
+
+			// Publish
+			publishButton: ( state: 'disabled' | 'enabled' ) => {
+				const buttonState = state === 'disabled' ? 'true' : 'false';
+				return `${ panel } button.editor-post-publish-button__button[aria-disabled="${ buttonState }"]`;
+			},
+
+			// List view
+			listViewButton: `${ panel } button[aria-label="List View"]`,
+
+			// Details popover
+			detailsButton: `${ panel } button[aria-label="Details"]`,
+
+			// Document Actions dropdown
+			documentActionsDropdown: `${ panel } button[aria-label="Show template details"]`,
+			documentActionsDropdownItem: ( itemSelector: string ) => `.popover-slot ${ itemSelector }`,
+			documentActionsDropdownShowAll: `.popover-slot .edit-site-template-details__show-all-button`,
+
+			// Editor settings
+			settingsButton: ( label = settingsButtonLabel ) =>
+				`${ panel } .edit-post-header__settings .interface-pinned-items button[aria-label="${ label }"]`,
+
+			// Undo/Redo
+			undoButton: 'button[aria-disabled=false][aria-label="Undo"]',
+			redoButton: 'button[aria-disabled=false][aria-label="Redo"]',
+
+			// More options
+			moreOptionsButton: ( label = moreOptionsLabel ) =>
+				`${ panel } button[aria-label="${ label }"]`,
+
+			// Site editor save
+			saveSiteEditorButton: `${ panel } button.edit-site-save-button__button`,
+
+			// Nav sidebar
+			navSidebarButton:
+				'button[aria-label="Block editor sidebar"],button[aria-label="Toggle navigation"]',
+		};
```

Comment:
**comment:blocking**: If I'm understanding correctly, this change is due to the changes introduced in Gutenberg 14.9.1 that adds an iframe within the editor field - is that correct?

If so, I would like to take a stab at fixing the issue at the foundational level before we resort to moving selectors inside the class definition, which is something we prescribe against in the [style guide](https://wpcalypso.wordpress.com/devdocs/test/e2e/docs/style_guide.md#no-selectors-within-class-definition).

Reply:
> If I'm understanding correctly, this change is due to the changes introduced in Gutenberg 14.9.1 that adds an iframe within the editor field - is that correct?

No. This is needed because Gutenberg 14.9.1 has introduced a refactored layout for the site editor, which is incompatible with the post editor layout. However, `EditorToolbarComponent` is used for both editors, so somehow we need to account for those different layouts. 

It's unrelated to the new iframe used for the post editor, since the previous selectors still work there.

Reply:
> the site editor [...] uses an editor toolbar with a DOM that is slightly different to the toolbar in the post editor

To expand on this, the main difference is that the container in the post editor is `.interface-interface-skeleton__header` while the container in the site editor is `.edit-site-layout__header`.

> moving selectors inside the class definition [...] is something we prescribe against in the [style guide](https://wpcalypso.wordpress.com/devdocs/test/e2e/docs/style_guide.md#no-selectors-within-class-definition).

I guess I can keep the selectors outside of the class, and transform all of them to functions that accept the container as parameter:

```
// ...
blockInserterButton: ( panel ) => `${ panel } button[class*="inserter-toggle"]`,
saveDraftButton: ( panel ) => `${ panel } button[aria-label="Save draft"], button[aria-label="Saved"]`,
switchToDraftButton: ( panel ) => `${ panel } button.editor-post-switch-to-draft`,
// ...
```

I honestly think the code is simpler by keeping the selectors within the class, so they can remain as plain strings, but happy to disagree and commit according to the style guide.

Reply:
> I guess I can keep the selectors outside of the class, and transform all of them to functions that accept the container as parameter

https://github.com/Automattic/wp-calypso/pull/71333/commits/4a7b231d94ecb1f975e950ff174d6e3254b7d28b